core/tests/puppeteer-acceptance-tests/specs/logged-out-user/generate-and-play-automated-voiceovers.spec.ts

```diff
@@ -0,0 +1,203 @@
+  let explorationId: string | null;
+
+  beforeAll(async function () {
+    curriculumAdmin = await UserFactory.createNewUser(
+      'curriculumAdm',
+      'curriculumAdmin@example.com',
+      [ROLES.CURRICULUM_ADMIN]
+    );
+
+    explorationEditor = await UserFactory.createNewUser(
+      'explorationEditor',
+      'exploration_editor@example.com'
+    );
+
+    releaseCoordinator = await UserFactory.createNewUser(
+      'releaseCoordinator',
+      'release_coordinator@example.com',
+      [ROLES.RELEASE_COORDINATOR]
+    );
+
+    voiceoverAdmin = await UserFactory.createNewUser(
+      'voiceoverAdm',
+      'voiceover_admin@example.com',
+      [ROLES.VOICEOVER_ADMIN]
+    );
+
+    await voiceoverAdmin.addSupportedLanguageAccentPair(
+      'English (United States)'
+    );
+    await voiceoverAdmin.enableAutogenerationForLanguageAccentPair('en-US');
+
+    // Enable the feature flag.
+    await releaseCoordinator.enableFeatureFlag(
+      'exploration_editor_can_modify_translations'
+    );
+    await releaseCoordinator.enableFeatureFlag(
+      'automatic_voiceover_regeneration_from_exp'
+    );
+
+    // Navigate to the creator dashboard and create a new exploration.
+    await explorationEditor.navigateToCreatorDashboardPage();
+    await explorationEditor.navigateToExplorationEditorPage();
+    await explorationEditor.dismissWelcomeModal();
+    await explorationEditor.updateCardContent(INTRODUCTION_CARD_CONTENT);
+    await explorationEditor.addInteraction(INTERACTION_TYPES.CONTINUE_BUTTON);
+
+    // Add the final card.
+    await explorationEditor.viewOppiaResponses();
+    await explorationEditor.directLearnersToNewCard(CARD_NAME.FINAL_CARD);
+    await explorationEditor.saveExplorationDraft();
+
+    await explorationEditor.navigateToCard(CARD_NAME.FINAL_CARD);
+    await explorationEditor.updateCardContent(
+      'We have practiced negative numbers.'
+    );
+    await explorationEditor.addInteraction(INTERACTION_TYPES.END_EXPLORATION);
+
+    // Navigate back to the introduction card and save the draft.
+    await explorationEditor.navigateToCard(CARD_NAME.INTRODUCTION);
+    await explorationEditor.saveExplorationDraft();
+
+    explorationId = await explorationEditor.publishExplorationWithMetadata(
+      'Test Exploration',
+      'This is a test exploration.',
+      'Algebra'
+    );
+    if (!explorationId) {
+      throw new Error('Error in publishing exploration successfully.');
+    }
+
+    await curriculumAdmin.createAndPublishTopic(
+      'Algebra I',
+      'Negative Numbers',
+      'Negative Numbers'
+    );
+
+    await curriculumAdmin.createAndPublishClassroom(
+      'Math',
+      'math',
+      'Algebra I'
+    );
+
+    await curriculumAdmin.createAndPublishStoryWithChapter(
+      'Algebra Story',
+      'algebra-story',
+      'Understanding Negative Numbers',
+      explorationId as string,
+      'Algebra I'
+    );
+
+    // Generating voiceovers to the exploration.
```

Comment:
Generate voiceovers for the exploration.

Reply:
Done